Userland/Applications/DisplaySettings/ThemesSettingsWidget.cpp

```diff
@@ -73,11 +72,11 @@ ErrorOr<void> ThemesSettingsWidget::setup_interface()
     auto mouse_settings_icon = TRY(Gfx::Bitmap::load_from_file("/res/icons/16x16/app-mouse.png"sv));
 
     m_color_scheme_names.clear();
-    Core::DirIterator iterator("/res/color-schemes", Core::DirIterator::SkipParentAndBaseDir);
-    while (iterator.has_next()) {
-        auto path = iterator.next_path();
-        TRY(m_color_scheme_names.try_append(TRY(String::from_deprecated_string(path.replace(".ini"sv, ""sv, ReplaceMode::FirstOnly)))));
-    }
+    Core::Directory::for_each_entry("/res/color-schemes"sv, Core::DirIterator::SkipParentAndBaseDir, [&](auto& entry, auto&) -> ErrorOr<IterationDecision> {
+        LexicalPath path { entry.name };
+        TRY(m_color_scheme_names.try_append(TRY(String::from_utf8(path.title()))));
+        return IterationDecision::Continue;
+    }).release_value_but_fixme_should_propagate_errors();
```

Comment:
Is there a reason to not just propagate this with `TRY()`? `ThemesSettingsWidget::setup_interface()` returns `ErrorOr<void>` anyways.

Reply:
Thanks, I missed this!